### PR TITLE
Executor2

### PR DESCRIPTION
**Pull request by arieivs**

### CODE REVIEW

File: includes/utils.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ void			free_redirs(t_redir *redirs, int nb_redirs);
 
 /* Error and/or Exit utils */
 t_error			error_and_exit(t_error code);
-t_error	        error_and_return(t_error code, int exit_code);
+int 	        error_and_return(t_error code, int exit_code);
```
Comment (Knulpinette) [inline includes/utils.h:37]: oh can you git pull from origin master ?
I mean i know those t_error are not necessary for the code to work but they're nice to know what type of value to expect =)

Comment (Knulpinette) [reply includes/utils.h:37]: (I changed those in my last little cleanup merge)

Comment (Knulpinette) [reply includes/utils.h:37]: Unless you changed it back to int on purpose :o

File: srcs/builtin_fts/env.c [hunk 2]
```diff
@@ -22,10 +22,10 @@ int	env(t_cmd_table *cmd, t_minishell *minishell)
...
 		if (get_env_lst_value(curr))
 		{
-			ft_putstr_fd(get_env_lst_name(curr), cmd->fd_in);
-			write(cmd->fd_in, "=", 1);
-			ft_putstr_fd(get_env_lst_value(curr), cmd->fd_in);
-			write(cmd->fd_in, "\n", 1);
+			ft_putstr_fd(get_env_lst_name(curr), cmd->fd_out);
+			write(cmd->fd_out, "=", 1);
+			ft_putstr_fd(get_env_lst_value(curr), cmd->fd_out);
+			write(cmd->fd_out, "\n", 1);
```
Comment (Knulpinette) [inline srcs/builtin_fts/env.c:28]: Why did you change everything to fd_out ? What was wrong ?

File: srcs/executing/executor.c [hunk 3]
```diff
@@ -43,6 +39,22 @@ static void	close_for_next_cmd(t_cmd_table cmd)
...
+// }
+// if (fd_out != 1)
+// {
+// 	dup2(fd_out, STDOUT_FILENO);
+// 	close(fd_out);
+// }
+// if (valid_command(pipex, 1)) // look into this function
+// 	execve(pipex->cmd_path, pipex->command, pipex->envp); // look into pipex struct
+// error_message(WRONG_CMD); // need to add this
+
```
Comment (Knulpinette) [inline srcs/executing/executor.c:57]: We'll need to brainstorm on the whole multi pipe thing. Because we need to go from scratch for pipex from what I understand. 
Maybe we can talk with some people about it once we have some ideas on how to deal with it ^^

File: srcs/parsing/parser.c [hunk 4]
```diff
@@ -34,7 +34,7 @@ t_error	parse(char *line)
 	minishell = get_minishell(NULL);
 	exit_code = lexer(line);
 	if (exit_code == STOP_EXECUTION)
-		return (error_and_return(0, STOP_EXECUTION));
+		return (STOP_EXECUTION);
```
Comment (Knulpinette) [inline srcs/parsing/parser.c:37]: haha fair point.

File: srcs/utils/error_exit.c [hunk 5]
```diff
@@ -15,7 +15,7 @@ t_error	error_and_exit(t_error code)
 	exit(EXIT_FAILURE);
 }
 
-t_error	error_and_return(t_error code, int exit_code)
+int	error_and_return(t_error code, int exit_code)
```
Comment (Knulpinette) [inline srcs/utils/error_exit.c:18]: oh ok so you did this on purpose. Ok !

File: includes/utils.h (hunk 1 above)
Comment (arieivs) [reply includes/utils.h:37]: I did 😅 cos that actually wasn't supposed to return a t_error, was really supposed to return an int. Most of the times I'm using it I'm returning 1, which is really supposed to be 1, not a t_error. And I do this cos what I return is usually the exit_code, and I store it and need it as such, which has no correspondance with our own homemade t_errors

File: srcs/builtin_fts/env.c (hunk 2 above)
Comment (arieivs) [reply srcs/builtin_fts/env.c:28]: That 😂 I was really stupid, I was writting to fd_in instead of fd_out 🤦‍♀️ and then of course it wasn't working

File: srcs/executing/executor.c (hunk 3 above)
Comment (arieivs) [reply srcs/executing/executor.c:57]: I was talking with Maria and José, they made me see something I wasn't counting with. I think I have an idea, just need to wrap my head around it and implement it.

File: srcs/utils/error_exit.c (hunk 5 above)
Comment (arieivs) [reply srcs/utils/error_exit.c:18]: Sry 😅 I should have discussed this in your PR instead of just changing it, but I noticed it only after I merged

File: includes/utils.h (hunk 1 above)
Comment (Knulpinette) [reply includes/utils.h:37]: ok !

File: srcs/builtin_fts/env.c (hunk 2 above)
Comment (Knulpinette) [reply srcs/builtin_fts/env.c:28]: hahahaha

File: srcs/executing/executor.c (hunk 3 above)
Comment (Knulpinette) [reply srcs/executing/executor.c:57]: oki doke

File: srcs/utils/error_exit.c (hunk 5 above)
Comment (Knulpinette) [reply srcs/utils/error_exit.c:18]: haha no problem, I figured I just wasn't sure =)
